### ISSUE TITLE
Syntax error

### ISSUE THREAD

**Issue by andya9** [mrichar1/clipster#15]
Following latest update I can’t run clipster because of:

```
line 438
    os.chmod(self.sock_file, 0600)

SyntaxError: invalid token
```
I tried changing to `600`: now the demon starts, but then when launching -s I get: `ERROR:Error connecting to socket. Is daemon running?`

**Comment by mrichar1** [comment]
Ahh - I hadn't realised that the use of octal numbers is deprecated and the correct way to do it is using constants from `stat`. python 2 still accepts `0600` but python 3 complains.

I've updated the code to do it the 'right' way - let me know if you still see any issues.

6e937a0052eac2cc2dba74a2ac938a1e96b9473b

**Comment by andya9** [comment]
The demon starts fine, but when trying -s i get:
```
Traceback (most recent call last):
  File "/home/arch/.shortcuts/clipster", line 631, in <module>
    main()
  File "/home/arch/.shortcuts/clipster", line 627, in main
    clipster.client_update(client_action)
  File "/home/arch/.shortcuts/clipster", line 52, in client_update
    'default_selection')))
TypeError: a bytes-like object is required, not 'str'
```

**Comment by mrichar1** [comment]
It looks like I had lost some of the utf-8 encoding of the messages being passed around when I refactored, and I had failed to properly test with both py2 and 3.  Obviously I need to put in some unit tests!

Should be fixed by 9ed4d3e830c41ae8ca7947be0b557a097e0947bb

**Comment by andya9** [comment]
Works perfectly now! Thank’you very much!
